### PR TITLE
Error creating factory partition with static DAC in secure cert partition (CON-749)

### PR DESCRIPTION
**Pull request by stonehouse**
We are looking at using the `mfg_tool` to generate factory partitions for ESP32c6. We are using pre-generated DAC certificates, and ran into some issues with the script. I've made the following changes and now the partitions look to be generated correctly.

We are using the following arguments:
`mfg_tool.py -cn x -v 0xFFFF -p 0xFFFF --pai -c pai.pem -cd cd.der --dac-cert dac_cert.pem --dac-key dac_key.pem --dac-in-secure-cert --target esp32c6 --size 0x3000 --vendor-name x --product-name 'x' --hw-ver 1 --hw-ver-str '1' --serial-num 1234567890 --csv factory_extra_schema.csv --mcsv extra_values.csv`

### CODE REVIEW

**Comment by dhrishi** [comment]
Thanks @stonehouse . @shubhamdp PTAL

File: tools/mfg_tool/mfg_tool.py [hunk 1]
```diff
@@ -244,7 +245,8 @@ def use_dac_from_args(args):
     logging.info('Generated DAC private key in binary format: {}'.format(out_private_key_bin))
     logging.info('Generated DAC public key in binary format: {}'.format(out_public_key_bin))
 
-    return out_cert_der, out_private_key_bin, out_public_key_bin
+    convert_private_key_from_pem_to_der(args.dac_key, out_private_key_der)
```
Comment (InfiniteYuan) [inline tools/mfg_tool/mfg_tool.py:248]: Why do we need to convert? The file is not used elsewhere.

File: tools/mfg_tool/mfg_tool.py [hunk 2]
```diff
@@ -375,7 +379,7 @@ def write_per_device_unique_data(args):
                                                  ca_cert = os.path.abspath(PAI['cert_der']), idf_target = args.target, 
                                                  op_file = secure_cert_partition_file_path)
 
-                append_cn_dac_to_csv(UUIDs[int(row['Index'])], os.sep.join([OUT_DIR['top'], UUIDs[int(row['Index'])], "internal", "DAC_cert.pem"]))
+                append_cn_dac_to_csv(UUIDs[int(row['Index'])], dac_cert_path)
```
Comment (InfiniteYuan) [inline tools/mfg_tool/mfg_tool.py:382]: We have fixed this issue internally and will synchronize it to github soon.

**Comment by InfiniteYuan** [comment]
[mfg tool](https://github.com/espressif/esp-matter/blob/main/tools/mfg_tool/mfg_tool.py#L386-L389) already support this feature，Please confirm and close it

**Pull request closed without merging**